bpo-45929: extend json.tool --json-lines to ignore empty rows

ZeeD opened the pull request:
As [45929](https://bugs.python.org/issue45929) is something that bugged me, I though it would be helpful to propose a PR.
Please let me know if this is good enough

<!-- issue-number: [bpo-45929](https://bugs.python.org/issue45929) -->
https://bugs.python.org/issue45929
<!-- /issue-number -->

ZeeD committed:
ignore empty lines for json_lines files

ZeeD committed:
add test_jsonlines_emptylines

ZeeD committed:
add test_jsonlines_emptylines

ZeeD committed:
fix typo

ZeeD committed:
Merge branch 'python:main' into master

blurb-it[bot] committed:
📜🤖 Added by blurb_it.

hongweipeng commented on Lib/json/tool.py:
```diff
@@ -15,6 +15,7 @@
 import sys
 from pathlib import Path
 
+_SENTINEL = object()
```
This variable seems to be unnecessary.

ZeeD replied:
I used an explicit sentinel because `json.loads(line)` can return `None`, an I needed a way to distinguish the cases ("do I need to call `json.dump` on `obj` or not?")
What approach do you suggest to avoid the sentinel?

hongweipeng replied:
The code `objs = (json.loads(line) if line.strip() else _SENTINEL for line in infile)` change to `objs = (json.loads(line) if line.strip() for line in infile)`,the effect is same.

ZeeD replied:
I think you mean `(json.loads(line) for line in infile if line.strip())` - as to just "ignore" empty rows in `objs`.
If I understand that well, then objs will not have any element on the newline.
Then, on #L75, on `for obj in objs: ... outfile.write('\n')` will not yield anything, not even an empty row, as output.
My desired outcome is to preserve those empty rows, so I used the `_SENTINEL`.
Or maybe are you suggesting something else?

hongweipeng replied:
Oh, I didn't notice that there is a `write('\n')` in the back.

ZeeD replied:
No problem :)
Given that, I think I can mark this conversation resolved? Feel free to open it again if you have some other inputs.

ZeeD committed:
Merge branch 'python:main' into master

ZeeD committed:
add newline at end of file

fix for https://dev.azure.com/Python/cpython/_build/results?buildId=97943&view=logs&j=4db1505a-29e5-5cc0-240b-53a8a2681f75&t=a975920c-8356-5388-147c-613d5fab0171&l=17

MaxwellDupre reviewed (approved):
Ran 16 tests in 3.341s
OK
Looks ok.

ZeeD committed:
Merge branch 'python:main' into master
